soc/nxp/kinetis/ke1xz/power.c

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2021 Vestas Wind Systems A/S
+ * Copyright (c) 2021, 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/pm/pm.h>
+#include <soc.h>
+
+LOG_MODULE_DECLARE(power, CONFIG_PM_LOG_LEVEL);
+
+#ifdef CONFIG_XIP
+__ramfunc static void wait_for_flash_prefetch_and_idle(void)
+{
+	uint32_t i;
+
+	for (i = 0; i < 8; i++) {
+		arch_nop();
+	}
+
+	k_cpu_idle();
+}
+#endif /* CONFIG_XIP */
+
+void pm_state_set(enum pm_state state, uint8_t substate_id)
+{
+	switch (state) {
+	case PM_STATE_RUNTIME_IDLE:
+		k_cpu_idle();
+		break;
+	case PM_STATE_SUSPEND_TO_IDLE:
+		/* Set partial stop mode and enable deep sleep */
+		SMC->STOPCTRL = SMC_STOPCTRL_PSTOPO(substate_id);
+		SCB->SCR |= SCB_SCR_SLEEPDEEP_Msk;
+#ifdef CONFIG_XIP
```

Comment:
is there a reason why you can't use if(IS_ENABLED()) and allow optimization instead of preprocessor directives (preprocessor directives tend to lead to configuration dependency problems and confusing issues in the long term evolution of C code, need to avoid when possible, even if it seems simple now)

Reply:
Thanks for the reminder, this looks better, I'll replace it.